File: js/axios/webhooks/deleteWebhook.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
  ? "https://api.theauthapi.com"
  : process.env.TESTING_URL;

async function deleteWebhook(webhookId) {
  try {
    const { data } = await axios
      .delete(`${apiUrl}/webhooks/${webhookId}`, {
        headers: {
          ContentType: "application/json",
          "x-api-key": accessKey,
        },
      });
    return data;
  } catch (error) {
    if (error.response) {
      console.log(error.response.data);
    } else {
      // handle other errors
    }
  }
}

(async () => {
  const deleted = await deleteWebhook(process.env.WEBHOOK_ID);
  console.log('Webhook has been deleted?', deleted);
})();

File: python/webhooks/delete_webhook.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/webhooks"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def delete_webhook(webhook_id):
    try:
        response = requests.delete(f'{api_url}/{webhook_id}', headers=headers)
        response.raise_for_status()
        return response.json()
    except HTTPError as error: 
        # handle error
        print(error)


webhook_id = 'REPLACE_WITH_YOUR_WEBHOOK_ID'
deleted = delete_webhook(webhook_id)
print('Webhook has been deleted?', deleted)
